services/app-web/src/pages/MccrsId/MccrsId.test.tsx

```diff
@@ -0,0 +1,141 @@
+import { screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import {
+    ldUseClientSpy,
+    renderWithProviders,
+} from '../../testHelpers/jestHelpers'
+import { MccrsId } from './MccrsId'
+
+describe('MCCRSID', () => {
+    beforeEach(() => {
+        ldUseClientSpy({ 'cms-questions': false })
+    })
+    afterEach(() => {
+        jest.resetAllMocks()
+    })
+
+    it('renders without errors', async () => {
+        renderWithProviders(<MccrsId />, {
+            routerProvider: {
+                route: '/submissions/15/MCCRS-record-number',
+            },
+        })
+
+        expect(
+            await screen.findByRole('heading', { name: 'MC-CRS record number' })
+        ).toBeInTheDocument()
+        expect(
+            screen.getByRole('button', { name: 'Save MC-CRS number' })
+        ).not.toHaveAttribute('aria-disabled')
+    })
+
+    it('displays the text field for mccrs id', async () => {
+        renderWithProviders(<MccrsId />, {
+            routerProvider: {
+                route: '/submissions/15/MCCRS-record-number',
+            },
+        })
+
+        expect(screen.getByTestId('textInput')).toBeInTheDocument()
```

Comment:
nit: Some of these tests could probably be combined into one. This text input could be moved to `'renders without errors'`. And the validation tests could be all in one test.